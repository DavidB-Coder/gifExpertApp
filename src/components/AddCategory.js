import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim()) {
            setCategory( ps => [inputValue,...ps] );        
            setInputValue('');
        }
    }


    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input
                    type="text"
                    value={ inputValue }                
                    onChange={(e) => {setInputValue(e.target.value)}}
                    placeholder={'Ingresa una nueva categorÃ­a'}
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategory : PropTypes.func.isRequired
};
