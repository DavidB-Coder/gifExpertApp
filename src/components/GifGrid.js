import React from 'react'
import { useFetchGifs } from '../Hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs( category );
    
    return (
        <>
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            <h3>{category}</h3>                        
            <div className="card-grid">
                {data.map((img) => {
                    return(
                        <GifGridItem 
                            key={img.id}                         
                            {...img}
                        />
                    )
                })}
            </div>            
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}