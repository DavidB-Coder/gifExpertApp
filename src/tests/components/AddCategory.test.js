import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('pruebas en el componente AddCategory', () => {
    
    const setCategory = jest.fn();
    let wrapper = shallow(<AddCategory setCategory={setCategory}/>);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategory={setCategory}/>);
    })
    
    test('debe mostrar el contenido del componente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('debe funcionar el mÃ©todo onChnage', () => {
        const value = 'Hola Mundo';
        const input = wrapper.find('input');
        input.simulate('change',{ target: {value}});

    });

    test('No debe postear la informacion con submit', () => {

        wrapper.find('form').simulate('submit',{ preventDefault(){} })
        expect( setCategory ).not.toHaveBeenCalled();

    })

    test('debe llamar el setcategories y limpiar la caja de texto', () => {
        const value = 'test'
        const input = wrapper.find('input');
        input.simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{ preventDefault(){} });
        expect( setCategory ).toHaveBeenCalled();
        expect( input.prop('value') ).toBe('');
        
    })
    
    
    
    

})
