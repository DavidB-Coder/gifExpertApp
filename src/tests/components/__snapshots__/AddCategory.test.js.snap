// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente AddCategory debe mostrar el contenido del componente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Ingresa una nueva categorÃ­a"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
